### PR TITLE
Remove some more vistir usage

### PR DESCRIPTION
**Pull request by oz123**
* Removed usage of vistir.compat.JSONDecodeError
  `json.JSONDecodeError` is Python 3.6+
* Removed vistir.path.rmtree in favour of shutil.rmtree
  Should work, but we might have to revert this.

### CODE REVIEW

**Comment by oz123** [comment]
@matteius I've seen this error for MacOS Python3.10 happen already a few times (in the main branch too). No idea how to fix it, but it seems unrelated to the changes I proposed in the PR.

File: pipenv/core.py [hunk 1]
```diff
@@ -2791,8 +2792,9 @@ def do_check(
 
 
 def do_graph(project, bare=False, json=False, json_tree=False, reverse=False):
+    import json as jsonlib
```
Comment (matteius) [inline pipenv/core.py:2795]: Can we just `import json` at the top level now?  I don't really see why to namespace it to `jsonlib` either.

File: pipenv/utils/shell.py [hunk 2]
```diff
@@ -252,17 +252,10 @@ def is_virtual_environment(path):
...
-        - already exists, silently complete
-        - regular file in the way, raise an exception
-        - parent directory(ies) does not exist, make them as well
-        From: http://code.activestate.com/recipes/82465-a-friendly-mkdir/
-    =======
-            - already exists, silently complete
-            - regular file in the way, raise an exception
-            - parent directory(ies) does not exist, make them as well
-            From: http://code.activestate.com/recipes/82465-a-friendly-mkdir/
-    >>>>>>> main
```
Comment (matteius) [inline pipenv/utils/shell.py:265]: Good catch

**Review by matteius — APPROVED** [review]

**Comment by matteius** [comment]
> @matteius I've seen this error for MacOS Python3.10 happen already a few times (in the main branch too). No idea how to fix it, but it seems unrelated to the changes I proposed in the PR.

I re-ran just the one failed job and it succeeded.  Looks like a 1-off GitHub data transfer failure

File: pipenv/utils/shell.py (hunk 2 above)
Comment (oz123) [reply pipenv/utils/shell.py:265]: pure luck :-)

**Pull request merged**